ethereum/contracts/SwapCreator.sol

```diff
@@ -1,61 +1,88 @@
 // SPDX-License-Identifier: LGPLv3
 pragma solidity ^0.8.19;
 
-import {IERC20} from "./IERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import {Secp256k1} from "./Secp256k1.sol";
 
+// SwapCreator facilitates swapping between Alice, a party that has ETH or an
+// ERC-20 token that she wants to exchange cross-chain for Monero (XMR), and
+// Bob, a party that has XMR and wishes to exchange it for Alice's ETH or token.
 contract SwapCreator is Secp256k1 {
-    // Swap state is PENDING when the swap is first created and funded
-    // Alice sets Stage to READY when she sees the funds locked on the other chain.
-    // this prevents Bob from withdrawing funds without locking funds on the other chain first
-    // Stage is set to COMPLETED upon the swap value being claimed or refunded.
+    using SafeERC20 for IERC20;
+
+    // The swap Stage is PENDING when `newSwap` is called to create and fund the
+    // swap. Alice sets Stage to READY, via `setReady`, after verifying that
+    // funds are locked on the other chain. Bob cannot claim the swap funds
+    // until Alice sets the swap Stage to READY. The Stage is set to COMPLETED
+    // when Bob claims directly via `claim` or indirectly via `claimRelayer`, or
+    // by Alice calling `refund`.
     enum Stage {
         INVALID,
         PENDING,
         READY,
         COMPLETED
     }
 
+    // swaps maps from a swap ID to the swap's current Stage
+    mapping(bytes32 => Stage) public swaps;
+
+    // Swap stores the initial swap configuration, the hash of which forms the
+    // swap ID.
     struct Swap {
-        // the swap initiator, Alice
-        // address allowed to refund the ether for this swap
+        // owner is the address of Alice, who initiates the swap by calling
+        // `newSwap`. Only the owner is allowed to call `setReady` or `refund`.
```

Comment:
```suggestion
        // `owner` is the address of the swap initiator (ie. the address which called `newSwap`).
        // Only the owner is allowed to call `setReady` or `refund`.
```

Reply:
Give that we now define "Alice" and "Bob" at the start of the contract, but we haven't defined "swap initiator" yet, doesn't it flow more logically the way it is?

Reply:
honestly i don't read code from top to bottom, so not really for me :p but not a huge deal